src/main/scala/io/iohk/ethereum/App.scala

```diff
@@ -1,59 +1,101 @@
 package io.iohk.ethereum
 
 import scala.concurrent.ExecutionContext.Implicits.global
-
-import akka.actor.ActorSystem
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
 import akka.agent._
 import io.iohk.ethereum.blockchain.sync.FastSyncController
 import io.iohk.ethereum.crypto._
 import io.iohk.ethereum.db.components.{SharedLevelDBDataSources, _}
 import io.iohk.ethereum.domain.{Blockchain, BlockchainImpl}
-import io.iohk.ethereum.network.{PeerManagerActor, ServerActor}
+import io.iohk.ethereum.network.PeerManagerActor.PeersResponse
+import io.iohk.ethereum.network.{PeerActor, PeerManagerActor, ServerActor}
 import io.iohk.ethereum.rpc.JsonRpcServer
 import io.iohk.ethereum.utils.{BlockchainStatus, Config, NodeStatus, ServerStatus}
+import org.spongycastle.crypto.AsymmetricCipherKeyPair
 
 object App {
 
-  import Config.{Network => NetworkConfig}
-
   val nodeKey = generateKeyPair()
 
-  val storagesInstance =  new SharedLevelDBDataSources with Storages.DefaultStorages
+  val storagesInstance: DataSourcesComponent with StoragesComponent = new SharedLevelDBDataSources with Storages.DefaultStorages
   val blockchain: Blockchain = BlockchainImpl(storagesInstance.storages)
 
   def main(args: Array[String]): Unit = {
     val actorSystem = ActorSystem("etc-client_system")
 
-    val nodeStatus =
-      NodeStatus(
-        key = nodeKey,
-        serverStatus = ServerStatus.NotListening,
-        blockchainStatus = BlockchainStatus(Config.Blockchain.genesisDifficulty, Config.Blockchain.genesisHash, 0))
+    val appActor = actorSystem.actorOf(AppActor.props(nodeKey, blockchain, storagesInstance, actorSystem), "app")
+
+    appActor ! AppActor.StartApp
+  }
+
+}
+
+class AppActor(nodeKey: AsymmetricCipherKeyPair,
+               blockchain: Blockchain,
```

Comment:
I cant complain about an effort to encapsulate but please leave all these as arguments. And leave the shutdownhook also, have a look here https://github.com/input-output-hk/etc-client/pull/68/files#diff-4238cdc00fed700564f7ead54ac6eb08R45 at how we'll wire these things together. At the top level we want the flexibility to wire the implementations together in various ways.